docs/source/class_basics.rst

```diff
@@ -267,6 +486,13 @@ be explicitly present if you are defining a protocol:
    # Error: nominal subtyping used by default
    x: NewProtocol = Concrete()  # Error!
 
+You can also include default implementations of methods in
+protocols. If you explicitly subclass these protocols you can inherit
+these default implementations. Explicitly including a protocol as a
+base class is also a way of documenting that your class implements a
+particular protocol, and it forces mypy to verify that your class
+implementation is actually compatible with the protocol.
```

Comment:
There is a caveat that this still works (although it shouldn't according to PEP 544):
```python
class P(Protocol):
    def meth(self) -> int: ...  # No error here...

class C(P):
    pass

C()  # ...and no error here!
```
mypy currently treats empty body (ellipsis) as a default implementation. There is an issue to allow this only in stubs (and  in tests), but it is not implemented yet. (The current workaround is to use ``@abstractmethod``, IIRC the similar problem with literal ellipsis in ``x: int = ...`` is already fixed.)

Reply:
For the reference here are the issues: https://github.com/python/mypy/issues/4066, https://github.com/python/mypy/issues/2350. You can increase priority if you think this is important.

Reply:
I don't think we need to call this out in the docs. I do think it's important to fix. (Honestly I don't understand the difference between the two issues you link to.)

Reply:
> (Honestly I don't understand the difference between the two issues you link to.)

They are duplicates, but both have some discussion, so I don't know which one to close.